numba/tests/test_function_type.py

```diff
@@ -1015,3 +998,33 @@ def foo_bad2good():
             str(cm.exception), r'.* function address is null')
 
         self.assertEqual(foo_bad2good(), 123)
+
+    def test_issue_5470(self):
+
+        @njit()
+        def foo1():
+            return 0
+
+        @njit()
+        def foo2():
+            return 0
+
+        formulae_foo = (foo1, foo1)
+
+        @njit()
+        def bar_scalar(_, __):
```

Comment:
Can we make this function uses `foo1` and `foo2` so that we know that the compiled function is correct?